### ISSUE TITLE
AutoOrientImage not working with IOS on iPhone

### ISSUE THREAD

**Issue by liotims** [kartik-v/bootstrap-fileinput#1269]
There is a issue with image orientation in preview. When I take a vertical photo from my iPhone and I upload it the picture is not oriented correctly
![capture](https://user-images.githubusercontent.com/3442191/42680825-69d4e01c-8686-11e8-8154-a5f651efaf4c.JPG)

In _previewFile function I see every time that the class "rorate-1" is added. It seem to have a issue to read EXIF informations of picture.

I use this workaround to solve this : 
Import [pexifjs script ](https://github.com/hMatoba/piexifjs)

and in fileinput.js I modified the _previewFile function like this : 
```
....
if ($img.length && self.autoOrientImage) {
	$h.validateOrientation(file, function (value) {
		if (!value) {
			self._validateImage(previewId, caption, ftype, fsize, iData);
			return;
		}
		
		var exifObj = piexif.load($img[0].href);
		value = exifObj["0th"][piexif.ImageIFD.Orientation];

		var $zoomImg = $preview.find('#zoom-' + previewId + ' img'), css = 'rotate-' + value;
		if (value > 4) {
			css += ($img.width() > $img.height() ? ' is-portrait-gt4' : ' is-landscape-gt4');
		}
		$h.addCss($img, css);
		$h.addCss($zoomImg, css);
		self._raise('fileimageoriented', {'$img': $img, 'file': file});
		self._validateImage(previewId, caption, ftype, fsize, iData);
		$h.adjustOrientedImage($img);
	});
}
....
```
jquery v 3.2.1
fileinput v 4.4.9
bootstrap v 4.1.1

**Comment by kartik-v** [comment]
Can you please submit a PR?

**Comment by liotims** [comment]
Done

**Comment by kartik-v** [comment]
I have enhanced this for various scenarios including appropriate scaling of images.

**Comment by shum16** [comment]
@kartik-v when are you planning to release a version fixing this issue? We do need it very much for our new project. Thank you.

**Comment by kartik-v** [comment]
The next release `v4.4.9` has been released.

**Comment by shum16** [comment]
Ok, loaded it separately via registerJsFile and it seems to work now on Chrome. We'll try on iPhone in a while..
Actually, I thought the widget should load piexifjs itself.

**Comment by kartik-v** [comment]
> Actually, I thought the widget should load piexifjs itself.

You are probably talking about the `yii2-widget-fileinput` extension. This is not related to this plugin. Any enhancement will be done there.

**Comment by shum16** [comment]
Ah, ok, I see )

**Comment by shum16** [comment]
Can't get it to work on iPhone... Anybody can confirm it's working there?

**Comment by kartik-v** [comment]
You may want to check [the image auto orientation demo page](http://plugins.krajee.com/file-image-management-demo#auto-image-orientation) and see if that works fine and cross check your configuration

**Comment by shum16** [comment]
Here's a screenshot of your demo site made on iPhone - the image is not rotated well.
![img_2720](https://user-images.githubusercontent.com/10737487/43309022-33daf490-918c-11e8-867b-853561c3266e.jpg)
By the way, do you process all jpeg extensions, including uppercase "JPG" and "jpeg". iPhone likes to save images with such extensions )

**Comment by shum16** [comment]
Your own example:
![img_2723](https://user-images.githubusercontent.com/10737487/43309228-dbb5db3a-918c-11e8-93b9-f3c1fd93943d.jpg)

**Comment by shum16** [comment]
@kartik-v do you have an idea?

**Comment by kartik-v** [comment]
@shum16  will check this as I get time... any info you can share will be helpful to debug

**Comment by shum16** [comment]
@kartik-v any new ion this?

**Comment by kartik-v** [comment]
@shum16 not got time to check this ... been busy... will try to see if I get time to look into this....

**Comment by Dannyson** [comment]
This is caused by mobile Safari. It autoorients pictures with EXIF information. So basically fileinput autoorients this picture, displays and Safari takes it, finds EXIF and rotates it again.

**Comment by kartik-v** [comment]
This will be resolved via enh #1362.
